app/src/main/java/com/sopt/now/utils/ApiFactory.kt

```diff
@@ -0,0 +1,39 @@
+package com.sopt.now.utils
+
+import com.jakewharton.retrofit2.converter.kotlinx.serialization.asConverterFactory
+import com.sopt.now.BuildConfig
+import com.sopt.now.datasource.AuthService
+import com.sopt.now.datasource.InfoService
+import kotlinx.serialization.json.Json
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.ResponseBody
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import java.util.concurrent.TimeUnit
+
+object ApiFactory {
+    private const val BASE_URL: String = BuildConfig.AUTH_BASE_URL
+
+    private val client: OkHttpClient = OkHttpClient.Builder()
+        .readTimeout(5000, TimeUnit.MILLISECONDS)
+        .connectTimeout(5000, TimeUnit.MILLISECONDS)
+        .addInterceptor(HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY))
+        .build()
+
+    val retrofit: Retrofit by lazy {
+        Retrofit.Builder()
+            .baseUrl(BASE_URL)
+            .client(client)
+            .addConverterFactory(Json.asConverterFactory("application/json".toMediaType()))
+            .build()
+    }
+
+    inline fun <reified T> create(): T = retrofit.create(T::class.java) // create를 통해서 retrofit 구현체 생성
+}
+
+object ServicePool {
+    val authService = ApiFactory.create<AuthService>()
+    val loginService = ApiFactory.create<AuthService>()
+    val infoService = ApiFactory.create<InfoService>()
+}
```

Comment:
```suggestion
object ServicePool {
    val authService: AuthService by lazy { create<AuthService>() }
    val loginService: AuthService by lazy { create<AuthService>() }
    val infoService: InfoService by lazy { create<InfoService>() }
}
```

이러면 서비스 중 하나만 사용되어도 ServicePool 객체가 참조될 때 나머지 서비스들도 모두 초기화되는 문제가 있습니다! 불필요한 리소스를 유발 할 것 같아요.

특히 네트워크 관련 서비스의 경우 초기화 과정에서 추가적인 시간이 소요될 수 있으므로 비효율적 일 수도 있을 것 같습니다!

Reply:
오 이런 생각은 못해봤는데!! 더 생각하면서 코드 작성을 해야겠네요..ㅎㅎ